bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/form/FragmentImpl.java

```diff
@@ -0,0 +1,184 @@
+
+    @OSGiService
+    private ModelFactory modelFactory;
+
+    @ValueMapValue(injectionStrategy = InjectionStrategy.OPTIONAL)
+    private String fragmentPath;
+
+    private String clientLibRef;
+
+    private Resource fragmentContainer;
+
+    @PostConstruct
+    private void initFragmentModel() {
+        ResourceResolver resourceResolver = resource.getResourceResolver();
+        fragmentContainer = null;
+        String fragmentRef = fragmentPath;
+        if (fragmentPath != null && StringUtils.contains(fragmentPath, "/content/dam/formsanddocuments")) {
+            fragmentRef = GuideUtils.convertFMAssetPathToFormPagePath(fragmentPath);
+        }
+        fragmentContainer = resourceResolver.getResource(fragmentRef + "/" + JcrConstants.JCR_CONTENT + "/guideContainer");
+        if (fragmentContainer != null) {
+            FormStructureParser formStructureParser = resource.adaptTo(FormStructureParser.class);
+            if (formStructureParser != null) {
+                this.clientLibRef = formStructureParser.getClientLibRefFromFormContainer();
+            }
+        }
+    }
+
+    @JsonIgnore
+    public String getFragmentPath() {
+        return fragmentPath;
+    }
+
+    @JsonIgnore
+    public String getClientLibRef() {
+        return clientLibRef;
+    }
+
+    @Override
+    public @NotNull Map<String, ? extends ComponentExporter> getExportedItems() {
+        if (itemModels == null) {
+            itemModels = getChildrenModels(request, ComponentExporter.class);
+        }
+        return itemModels;
+    }
+
+    protected <T> Map<String, T> getChildrenModels(@Nullable SlingHttpServletRequest request, @NotNull Class<T> modelClass) {
+        Map<String, T> models = new LinkedHashMap<>();
+        List<Resource> filteredChildrenResources = getFilteredChildrenResources(fragmentContainer);
+        SlingHttpServletRequest wrappedSlingHttpServletRequest = new SlingHttpServletRequestWrapper(request) {
+
+            @Override
+            public Object getAttribute(String attrName) {
+                if (FormConstants.RESOURCE_CALLER_PATH.equals(attrName)) {
+                    String resourceCallerPath = (String) super.getAttribute(FormConstants.RESOURCE_CALLER_PATH);
+                    // If the attribute is already defined then we're in a nested situation.
+                    // The code for computing the components id uses the root-most resource path
+                    // (because of how componentContext is handled in HTML rendering, so we return
+                    // that.
+                    return (resourceCallerPath != null) ? resourceCallerPath : resource.getPath();
+                }
+                return super.getAttribute(attrName);
+            }
+        };
+        for (Resource child : filteredChildrenResources) {
+            T model = null;
+            if (request != null) {
+                // todo: if possible set i18n form parent to child here, this would optimize the first form rendering
```

Comment:
@ujjwal5 this will need to be passed from the context in which form is being rendered (that should take care of both the flows from standalone form and form embedded in site page)